test-common/test-common-util/src/main/java/org/apache/kafka/common/test/junit/QuarantinedPostDiscoveryFilter.java

```diff
@@ -41,42 +41,62 @@ public class QuarantinedPostDiscoveryFilter implements PostDiscoveryFilter {
 
     public static final String RUN_QUARANTINED_PROP = "kafka.test.run.quarantined";
 
+    public static final String RUN_FLAKY_PROP = "kafka.test.run.flaky";
+
     public static final String CATALOG_FILE_PROP = "kafka.test.catalog.file";
 
     private final Filter<TestDescriptor> autoQuarantinedFilter;
+
     private final boolean runQuarantined;
 
+    private final boolean runFlaky;
+
     // No-arg public constructor for SPI
     @SuppressWarnings("unused")
     public QuarantinedPostDiscoveryFilter() {
         runQuarantined = System.getProperty(RUN_QUARANTINED_PROP, "false")
             .equalsIgnoreCase("true");
 
+        runFlaky = System.getProperty(RUN_FLAKY_PROP, "false")
+            .equalsIgnoreCase("true");
+
         String testCatalogFileName = System.getProperty(CATALOG_FILE_PROP);
         autoQuarantinedFilter = AutoQuarantinedTestFilter.create(testCatalogFileName, runQuarantined);
     }
 
     // Visible for tests
-    QuarantinedPostDiscoveryFilter(Filter<TestDescriptor> autoQuarantinedFilter, boolean runQuarantined) {
+    QuarantinedPostDiscoveryFilter(
+        Filter<TestDescriptor> autoQuarantinedFilter,
+        boolean runQuarantined,
+        boolean runFlaky
+    ) {
         this.autoQuarantinedFilter = autoQuarantinedFilter;
         this.runQuarantined = runQuarantined;
+        this.runFlaky = runFlaky;
     }
 
     @Override
     public FilterResult apply(TestDescriptor testDescriptor) {
         boolean hasTag = testDescriptor.getTags().contains(FLAKY_TEST_TAG);
         FilterResult result = autoQuarantinedFilter.apply(testDescriptor);
-        if (runQuarantined) {
-            // If selecting quarantined tests, we first check for explicitly flaky tests. If no
-            // flaky tag is set, check the auto-quarantined filter. In the case of a missing test
-            // catalog, the auto-quarantined filter will exclude all tests.
+
+        if (runFlaky && runQuarantined) {
```

Comment:
Pardon me, in the `build.yaml` we exclude the condition `run-flaky: true and run-new: true`, so is this a unreachable path?

Reply:
It is unreachable with our current configuration, but since it is possible to hit this I figured we should support it in code.